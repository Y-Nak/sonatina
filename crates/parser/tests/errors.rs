use std::path::Path;

use dir_test::{dir_test, Fixture};
use sonatina_parser::parse_module;
mod common;

#[dir_test(
    dir: "$CARGO_MANIFEST_DIR/test_files/errors/",
    glob: "*.sntn"
)]
fn test_errors(fixture: Fixture<&str>) {
    let Err(errs) = parse_module(fixture.content()) else {
        panic!("expected parse_module to fail with errors");
    };
    let path = Path::new(fixture.path())
        .file_name()
        .unwrap()
        .to_string_lossy();

    let mut v = vec![];
    for err in errs {
        err.print(&mut v, &path, fixture.content(), false).unwrap();
    }
    let s = String::from_utf8(v).unwrap();
    snap_test!(s, fixture.path());
}
