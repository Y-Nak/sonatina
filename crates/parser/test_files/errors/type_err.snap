---
source: crates/parser/tests/errors.rs
expression: s
input_file: crates/parser/test_files/errors/type_err.sntn
---
error: parse error
 --> type_err.sntn:4:12
  |
4 |         v0.i16 = add 2.i8 3.i8;
  |            ^^^ type mismatch: value declared as `i16`, but inferred type is `i8`
  |error: parse error
 --> type_err.sntn:5:12
  |
5 |         v1.i8 = call %foo;
  |            ^^ type mismatch: value declared as `i8`, but inferred type is `i16`
  |
